mithril-signer/src/message_adapters/to_register_signer_message.rs

```diff
@@ -1,26 +1,44 @@
+use anyhow::Context;
 use mithril_common::{
     entities::{Epoch, Signer},
-    messages::{RegisterSignerMessage, ToMessageAdapter},
+    messages::{RegisterSignerMessage, TryToMessageAdapter},
+    StdResult,
 };
 
 /// Adapter to create [RegisterSignerMessage] from [Signer] instance.
 pub struct ToRegisterSignerMessageAdapter;
 
-impl ToMessageAdapter<(Epoch, Signer), RegisterSignerMessage> for ToRegisterSignerMessageAdapter {
+impl TryToMessageAdapter<(Epoch, Signer), RegisterSignerMessage>
+    for ToRegisterSignerMessageAdapter
+{
     /// Method to trigger the conversion.
-    fn adapt((epoch, signer): (Epoch, Signer)) -> RegisterSignerMessage {
-        RegisterSignerMessage {
+    fn try_adapt((epoch, signer): (Epoch, Signer)) -> StdResult<RegisterSignerMessage> {
+        Ok(RegisterSignerMessage {
```

Comment:
wouldn't it be more readable if we assign the `RegisterSignerMessage` to a binding then return a `Ok(message)` instead of writing everything inline?